elyra/pipeline/parser.py

```diff
@@ -86,6 +86,17 @@ def parse(pipeline_definition) -> Pipeline:
                     outputs=node['app_data'].get('outputs') or [],
                     dependencies=links
                 )
+                # validate that the operation has all required properties
```

Comment:
Should the validation be enforced during construction?

Reply:
what do you mean, in the constructor of the operation?